state/protocol/badger/state_root.go

```diff
@@ -0,0 +1,125 @@
+package badger
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// StateRoot is the root information required to bootstrap the protocol state
+type StateRoot struct {
+	block          *flow.Block
+	result         *flow.ExecutionResult
+	seal           *flow.Seal
+	epochFirstView uint64
+}
+
+func NewStateRoot(block *flow.Block, result *flow.ExecutionResult, seal *flow.Seal, epochFirstView uint64) (*StateRoot, error) {
+	err := validate(block, result, seal, epochFirstView)
+	if err != nil {
+		return nil, fmt.Errorf("inconsistent state root: %w", err)
+	}
+	return &StateRoot{
+		block:          block,
+		result:         result,
+		seal:           seal,
+		epochFirstView: epochFirstView,
+	}, nil
+}
+
+// Validate checks internal consistency of state root
+func validate(block *flow.Block, result *flow.ExecutionResult, seal *flow.Seal, epochFirstView uint64) error {
```

Comment:
comment for other reviewers:
* checking the validity of the provided bootstrapping information was previously part of the mutators `Bootstrap` method (see [here](https://github.com/onflow/flow-go/blob/115ccee9cae920ee5bc309537cf43a9b152a1cf9/state/protocol/badger/mutator.go#L24-L73)) is not performed by the `StateRoot`: https://github.com/onflow/flow-go/blob/32b363388cc2aca861d3a46d23a2e7a1ed801377/state/protocol/badger/state_root.go#L30-L31
* encapsulating the required bootstrapping information (we will probably require more for Epochs) as well as checking its validity is now the responsibility of `StateRoot`

Reply:
The StateRoot seems to be a new module, as well as this validate function. Does this validate function exist before?  Did we change any validation logic here or simply moving around the code?

I was going through each checks, but not sure if necessary.